homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,234 @@
+
+PARALLEL_UPDATES = 0
+
+
+class PowerSupplyMode(StrEnum):
+    """LED bar mode."""
+
+    BATTERY = "battery"
+    EXTERNAL = "external"
+    BATTERY_EXTERNAL = "battery+external"
+
+
+@dataclass(kw_only=True, frozen=True)
+class WatergateSensorEntityDescription(SensorEntityDescription):
+    """Description for Watergate sensor entities."""
+
+    value_fn: Callable[
+        [WatergateAgregatedRequests],
+        str | int | float | datetime | PowerSupplyMode | None,
+    ]
+
+
+DESCRIPTIONS: list[WatergateSensorEntityDescription] = [
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.water_meter.duration
+        if data.state and data.state.water_meter
+        else None,
+        translation_key="water_meter_volume",
+        key="water_meter_volume",
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.water_meter.duration
+        if data.state and data.state.water_meter
+        else None,
+        translation_key="water_meter_duration",
+        key="water_meter_duration",
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.ip if data.networking else None,
+        translation_key="ip",
+        key="ip",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.gateway if data.networking else None,
+        translation_key="gateway",
+        key="gateway",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.subnet if data.networking else None,
+        translation_key="subnet",
+        key="subnet",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.ssid if data.networking else None,
+        translation_key="ssid",
+        key="ssid",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.rssi if data.networking else None,
+        translation_key="rssi",
+        key="rssi",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
```

Comment:
You can remove the translation key here and use the device class translating making it automatically be named `Signal strength`